bot/exts/backend/error_handler.py

```diff
@@ -15,6 +18,51 @@
 log = get_logger(__name__)
 
 
+class CheckedDeleteMessageButton(DeleteMessageButton):
+    """Button to delete a message, restricted to the message owner and moderators."""
+
+    def __init__(self, owner: User | Member):
+        super().__init__()
+        self.owner = owner
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Delete the message if the user is the owner of the message or a moderator."""
+        user = interaction.user
+        if (
+            user.id == self.owner.id or
+            (
+                isinstance(user, Member) and
+                any(user.get_role(role) for role in MODERATION_ROLES)
+            )
+        ):
+            await super().callback(interaction)
+        else:
+            await interaction.response.send_message("You can only delete your own command responses!", ephemeral=True)
+
+
+class HelpEmbedView(discord.ui.View):
+    """View to allow showing the help command for command error responses."""
+
+    def __init__(self, help_embed: Embed, owner: User | Member):
+        super().__init__()
+        self.help_embed = help_embed
+        self.message: Message | None = None
+
+        self.delete_button = CheckedDeleteMessageButton(owner)
+        self.add_item(self.delete_button)
+
+    @discord.ui.button(label='Help', style=ButtonStyle.primary)
+    async def help_button(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Send an ephemeral message with the contents of the help command."""
+        await interaction.response.send_message(embed=self.help_embed, ephemeral=True)
+
+    async def on_timeout(self) -> None:
+        """Remove the view from `self.message` if set."""
+        if self.message:
```

Comment:
I'm curious, will we ever fall into this case where `self.message` is None ? 
The timer will start when the view is sent, and it cannot timeout in the meantime (I tried putting timeout to 1 second and an asyncio.sleep(5) before sending it :P just to verify), so view.message will always be present since that's not something that can fail.

If what I'm saying is true, then the `if self.message` check can be removed

Reply:
I think you're probably correct that it shouldn't ever be `None` with the current code. I mainly put the check there because it's more obviously correct since `self.message` has the type `Message | None`, and it doesn't do any harm to check.

Reply:
Sure sure, which is why I said that I was only curious to know. Not that there's an issue with the way it is or anything